tools-local/unix/invoke-extension.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "usage: $0 <script-name> <repository-root> <tool-name> <override-scripts-folder-path> <strict-tool-version-match> <tool-path>"
+    echo "  script-name: Name of the extension script."
+    echo "  repository-root: Path to repository root."
+    echo "  tool-name: Name of the tool."
+    echo "  override-scripts-folder-path: If a path is specified then, search and acquire scripts from the specified folder will be invoked."
+    echo "                                  Otherwise, search will use the default search and acquire scripts located within the repository."
+    echo "  strict-tool-version-match: If equals to \"strict\" then, search will ensure that the version of the tool searched is the declared version."
+    echo "                                          Otherwise, search will attempt to find a version of the tool, which may not be the declared version."
+    echo "  tool-path: Path to the tool executable."
```

Comment:
Doesn't this script just pass on `strict-tool-version-match` and `tool-path`? And don't we need to call some extensible scripts without knowing the `tool-path` yet? I think I've mentioned that I expect some sort of `[args ...]` notation.